src/daemon/ca/status.rs

```diff
@@ -164,107 +245,225 @@ impl StatusStore {
+    // where
+    //     F: FnOnce(&mut ChildStatus),
+    // {
+    //     self.update_ca_status(ca, |status| {
+    //         let child_status = match status.children.get_mut(child) {
+    //             Some(child_status) => child_status,
+    //             None => {
+    //                 status.children.insert(child.clone(), ChildStatus::default());
+    //                 status.children.get_mut(child).unwrap()
+    //             }
+    //         };
+    //         op(child_status)
+    //     })
+    //     .await
+    // }
+
+    // fn update_status<F>(&self, ca: &Handle, op: F) -> KrillResult<()>
+    // where
+    //     F: FnOnce(&mut CaStatus) -> KrillResult<()>,
+    // {
+    //     let cache = self.cache.write().unwrap();
+
+    //     if !cache.contains_key(ca) {
+    //         cache.insert(ca.clone(), Arc::new(CaStatus::default()));
+    //     }
+
+    //     let mut ca_status = cache.get_mut(ca).unwrap(); // safe, we just set it if missing
+    //     let mut_status = Arc::make_mut(&mut ca_status);
+
+    //     op(&mut mut_status)?;
+
+    //     Ok(())
+    // }
+
+    // async fn update_ca_status<F>(&self, ca: &Handle, op: F) -> KrillResult<()>
+    // where
+    //     F: FnOnce(&mut CaStatus),
+    // {
+    //     let lock = self.locks.ca(ca).await;
+    //     let _ = lock.write().await;
+
+    //     let mut status: CaStatus = self
+    //         .store
+    //         .get(&Self::full_status_key(ca))
+    //         .ok()
+    //         .flatten()
+    //         .unwrap_or_default();
+
+    //     op(&mut status);
+
+    //     self.store.store(&Self::full_status_key(ca), &status)?;
+
+    //     Ok(())
+    // }
+
     fn error_to_error_res(error: &Error) -> ErrorResponse {
         match error {
             Error::HttpClientError(httpclient::Error::ErrorResponseWithJson(_, _, res)) => res.clone(),
             _ => error.to_error_response(),
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+
+    use std::path::PathBuf;
+
+    use crate::commons::util::file;
+    use crate::test::test_under_tmp;
+
+    #[test]
+    fn read_save_status() {
+        test_under_tmp(|d| {
+            let source = PathBuf::from("test-resources/status_store/migration-0.9.5/");
+            let target = d.join("status");
+            file::backup_dir(&source, &target).unwrap();
+
+            let status_testbed_before_migration =
+                include_str!("../../../test-resources/status_store/migration-0.9.5/testbed/status.json");
+
+            let status_testbed_before_migration: CaStatus =
+                serde_json::from_str(status_testbed_before_migration).unwrap();
+
+            let store = StatusStore::new(&d, "status").unwrap();
+            let testbed = Handle::from_str("testbed").unwrap();
+
+            let status_testbed_migrated = store.get_ca_status(&testbed);
+
+            assert_eq!(&status_testbed_before_migration, status_testbed_migrated.as_ref());
```

Comment:
What is this actually testing? If the migration did nothing this would still succeed wouldn't it?

Reply:
No it wouldn't. This tests that the CaStatus is the same when parsed before doing any migration, and after it has been migrated. The 'hidden' knowledge in the test is that the original status.json is parsed, split into components, and that these small components are then parsed.